### ISSUE TITLE
End of test run fails: can't merge directories

### ISSUE THREAD

**Issue by pitrou** [conda/conda-build#1375]
With conda-build 2.0.1, I get the following error at the end of a test run (`conda build --test`):
```
OSError: Can't merge C:\Users\buildbot\hpa-ci\buildbot-slave\var-64\llvmlite-py2_7-w64\build into C:\conda64\conda-bld\llvmlite_1474303796131\work: file exists: C:\conda64\conda-bld\llvmlite_1474303796131\work\.codeclimate.yml
```

I don't really understand what's happening here.

**Comment by msarahan** [comment]
I'm not sure exactly where the error is coming from, but maybe I can explain the "merge" part.

In an effort to support (or rather, at least not disable) namespace packages, we merge folders, rather than copying them.  This is just to say that when copying them, we allow existing folders to be written into.  We throw an error, however, if any file being written already exists.  So, you're probably seeing this.  

I am guessing that it might be happening because your .codeclimate.yml file is (accidentally?) being packaged, and/or you have specified ```test/source_files``` in meta.yaml in such a way that you are copying more than you probably should be.

**Comment by pitrou** [comment]
We don't have any `test/source_files` in meta.yaml.
When does folder merging happen exactly? By the sound of it, it should happen at install time ("to support namespace pacakges")? Why did `conda-build --test` fail there?

**Comment by msarahan** [comment]
Sorry for the delay here.  Can you post more of the traceback?

**Comment by msarahan** [comment]
I encountered this error, and believe I have fixed it in the 2.0.4 release.  I'm closing this issue.  If you still see it, please reopen or open a new issue.
